contracts/NonfungibleTokenPositionDescriptor.sol

```diff
@@ -11,7 +12,21 @@ import './libraries/NFTDescriptor.sol';
 
 /// @title Describes NFT token positions
 /// @notice Produces a string containing the data URI for a JSON metadata string
-contract NonfungibleTokenPositionDescriptor is INonfungibleTokenPositionDescriptor {
+contract NonfungibleTokenPositionDescriptor is INonfungibleTokenPositionDescriptor, Ownable {
+    // tokens that take priority as ratio numerator (such as stablecoins)
+    mapping(address => bool) public ratioNumeratorTokens;
+    // tokens that take priorty as ratio denominator (such as WETH)
+    mapping(address => bool) public ratioDenominatorTokens;
+
+    constructor(address[] memory numeratorTokens, address[] memory denominatorTokens) Ownable() {
```

Comment:
this state could complicate the upgradeability

any way to do this with immutability, i.e. so that it's in the code? tbh i don't know of one but if there's another strategy would be great

Reply:
I guess I imagined it might be something where we'd want to add more as we saw fit. Though having governance call the implementation is a tiny bit tricky per this comment (https://github.com/Uniswap/uniswap-v3-periphery/pull/60/files#r605287205)

why would this state complicated upgradeability?